WebGoatCore/Models/CreditCard.cs

```diff
@@ -34,34 +35,50 @@ public void GetCardForUser()
             XDocument document = ReadCreditCardFile();
             try
             {
-                var xElement = document.Descendants("CreditCard").FirstOrDefault(c => c.Element("Username").Value.Equals(Username));
-                if (!xElement.HasElements)
+                XElement ccElement = GetCreditCardXmlElement(document);
+
+                Username = Username;
```

Comment:
Is this assignment correct? I think something like `Username = ccElement.Element("Username")` was intended.

Reply:
Right. Fixed.